pruner/light/pruner.go

```diff
@@ -3,15 +3,49 @@ package light
 import (
 	"context"
 
+	"github.com/ipfs/boxo/blockservice"
+	"github.com/ipfs/boxo/blockstore"
+	"github.com/ipfs/go-datastore"
+
 	"github.com/celestiaorg/celestia-node/header"
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/ipld"
 )
 
-type Pruner struct{}
+type Pruner struct {
+	bserv blockservice.BlockService
+	ds    datastore.Datastore
+}
 
-func NewPruner() *Pruner {
-	return &Pruner{}
+func NewPruner(bstore blockstore.Blockstore, ds datastore.Datastore) *Pruner {
+	return &Pruner{bserv: blockservice.New(bstore, nil), ds: ds}
 }
 
-func (p *Pruner) Prune(context.Context, *header.ExtendedHeader) error {
+func (p *Pruner) Prune(ctx context.Context, h *header.ExtendedHeader) error {
+	dah := h.DAH
+	if share.DataHash(dah.Hash()).IsEmptyRoot() {
+		return nil
+	}
+
+	var roots [][]byte
+	roots = append(roots, h.DAH.RowRoots...)
+	roots = append(roots, h.DAH.ColumnRoots...)
+	for _, root := range roots {
+		cid := ipld.MustCidFromNamespacedSha256(root)
+		if err := ipld.DeleteNode(ctx, p.bserv, cid); err != nil {
+			return err
+		}
+	}
+
+	key := rootKey(dah)
+	err := p.ds.Delete(ctx, key)
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
+
+func rootKey(root *share.Root) datastore.Key {
```

Comment:
I thought we had some convenience func for that in share pkg

we also have it in light/availability so maybe we can dedup

Reply:
We touched upon this with Vlad. The core issue is that availability and pruning are detached. Should be fixed by https://github.com/celestiaorg/celestia-node/issues/3394